### DISCUSSION TITLE
EDLogging returns ERROR for qualname when called out of callback?

### DISCUSSION THREAD

**Discussion by huehnerfaust** [EDCD/EDMarketConnector#1662] [Plugin Development]
Hi,

please excuse me not knowing enough of scope in python. Its my first project in python.

So i have set up a plugin which displays some ttk.optionmenu's. When i call the logger out of the callback of one of my optionsmenu (when a user selects a new option), i get that following error in the market connector log. 

ALERT!  Something went wrong with finding caller qualname for logging!
2022-09-13 15:46:29.676 UTC - INFO - 3300:9312:9312 <plugins>.BackpackLog.load.<ERROR in EDMCLogging.caller_class_and_qualname() for "qualname">:61:

plus my formated log string after the line number.

My formated log string is displayed correctly!

So its not realy breaking anythink. But i'm afraid of any side effects and realy wan't to understand where the problem is.

Edit: i followed plugin.rm to setup the logging functionality, works fine, beside my callsbacks

**Comment by Athanasius** [comment]
This should be a [Bug Report](https://github.com/EDCD/EDMarketConnector/issues/new?assignees=&labels=bug%2C+unconfirmed&template=bug_report.md&title=) - not a Discussion, so that it fits into our workflow better.

When opening the Bug Report please supply a minimum plugin example that triggers the ERROR so we can reproduce and better investigate.

**Comment by huehnerfaust** [comment]
Thanks for your reply.

I think its more a case of me not understanding python quite right.

After more fiddling with my code, i can get the "error" to disapear.  When i remove the "global variablesnames, ..." keyword in my call back function (and just use my outside variables instead of expanding the callback scope, this is the part which confuses me the most i think), everythink works and prints as expected.

So i realy think its not a bug on the log implementation side. More like me still learning the workings of python. Excuse my stubornes =)


I realy like market connector, its a great peace of software and i use it all the time when playing. At first i thought i code myself a quick and dirty nodejs app to survey the log file, but figured by writing a plugin i could learn some python and maybe give somethink back to this community. Thanks to you for supporting edmc!
